README.md

```diff
@@ -196,13 +196,62 @@ knife ec2 server create -I ami-d0f89fb9 --ssh-key your-public-key-id -f m1.mediu
+# Possible values of --server-connect-attribute: private_dns_name, private_ip_address, dns_name, public_ip_address
 # If --server-connect-attribute is not specified, knife attempts to determine if connecting to the instance's public or private IP is most appropriate based on other settings
 knife ec2 server create -I ami-173d747e -x ubuntu --server-connect-attribute public_ip_address
 ```
 
 View additional information on configuring Windows images for bootstrap in the documentation for [knife-windows](https://docs.chef.io/plugin_knife_windows.html).
 
+
+#### Bootstrap Windows instance without user-data through winrm ssl transport
+
+Users can bootstrap the Windows instance without the need to provide the user-data. `knife-ec2` has the ability to bootstrap the Windows instance through `winrm protocol` using the `ssl` transport. This require users to set `--winrm-transport` option as `ssl`, `--winrm-authentication-protocol` option as `negotiate` and `--winrm-ssl-verify-mode` option as `verify_none`. This will reduce the users overhead as they would not be required to pass any user-data to configure the winrm for bootstrapping the instance.
+
+***Note***: Users also need to pass the `security_group` with having the required ports opened like `5986` for `winrm ssl transport`.
+
+Below is the sample command to create a Windows instance and bootstrap it through `ssl` transport without passing any user-data:
+
+```
+knife ec2 server create -N chef-node-name -I your-windows-image -f flavor-of-server -x '.\a_local_user' -P 'yourpassword' --ssh-key your-public-key-id --winrm-transport ssl --winrm-authentication-protocol negotiate --winrm-ssl-verify-mode verify_none --security-group-ids your-security-groups -VV
+```
+
+#### Bootstrap Windows instance with user-data through winrm plaintext transport
+
+Users can bootstrap the Windows instance through `winrm plaintext transport` as well. But for this users need to pass the user-data along with `--winrm-transport` option as `plaintext` and `--winrm-authentication-protocol` option as `basic`.
+
+***Note***: Users also need to pass the `security_group` with having the required ports opened like `5985` for `winrm plaintext transport`.
+
+Below is the sample user-data that users can use for `plaintext` transport:
+
+```
+<powershell>
+
+$user=<user_name>
+$password=<password>
+
+net user /add $user $password;
+net localgroup Administrators /add $user;
+
+winrm quickconfig -q
+winrm create winrm/config/Listener?Address=*+Transport=HTTP
+winrm set winrm/config/service '@{AllowUnencrypted="true"}'
+winrm set winrm/config/service/auth '@{Basic="true"}'
+winrm set winrm/config/winrs '@{MaxMemoryPerShellMB="300"}'
+winrm set winrm/config '@{MaxTimeoutms="1800000"}'
+
+netsh advfirewall firewall set rule name="Windows Remote Management (HTTP-In)" profile=public protocol=tcp localport=5985 remoteip=localsubnet new remoteip=any
+
+</powershell>
+```
```

Comment:
I would change the winrm settings to something like:

```
### Windows 2008R2 ONLY ##########
winrm quickconfig -q
winrm set winrm/config/winrs '@{MaxMemoryPerShellMB="1024"}'
 ```

The firewall change should definitely stay.

Now that winrm/knife-windows supports cross platform negotiate auth we should stop publicizing basic auth and allowing unencrypted traffic. Also Windows 2012 and higher has winrm turned on by default and has a much higher memory limit. In fact, explicitly enabling winrm on a 2012 box is counter productive because it restarts the service and its possible that the converge is already in progress.